ccc/tests/test_parameters.py

```diff
@@ -13,70 +13,72 @@ def test_update_specification_with_dict():
     cyr = 2020
     spec = Specification(year=cyr)
     new_spec_dict = {
-        'profit_rate': {cyr: 0.4},
-        'm': {cyr: 0.5}
+        'profit_rate': 0.4,
+        'm': 0.5
     }
     spec.update_specification(new_spec_dict)
     assert spec.profit_rate == 0.4
     assert spec.m == 0.5
-    assert len(spec.parameter_errors) == 0
+    assert len(spec._errors) == 0
 
 
 def test_update_specification_with_json():
     cyr = 2020
     spec = Specification(year=cyr)
     new_spec_json = """
     {
-        "profit_rate": {"2020": 0.4},
-        "m": {"2020": 0.5}
+    "profit_rate": {"value": [{"year": 2020, "value": 0.4}]},
```

Comment:
@hdoupe Can you advise on the formatting of the `new_spec_json` here?  I feel like I'm following the format used in the `default_parameters.json`, but I'm getting the following error:
```
self = <json.decoder.JSONDecoder object at 0x107b82588>, s = 'revision', idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.

        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.

        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/anaconda3/lib/python3.7/json/decoder.py:355: JSONDecodeError
```

Thanks!

Reply:
The JSON should be written like this. You can update the data with a dict too. It doesn't have to be JSON:

```python
new_spec_json = {
    "profit_rate": [
        {
            "year": 2020,
            "value": 0.4
        }
    ],
    "m": [
        {
            "year": 2020,
            "value": 0.5
        }
    ]
}

```

Then you can update your params like this:

```python
spec.adjust(new_spec_json)
spec.compute_default_params()

```

You could wrap these method calls in your `Specification.update_specification` method.

Reply:
@hdoupe , These methods are wrapped in `Specification.update_specification`.

I have a test of a reform entered as a dict.  I wanted to also have test with a JSON format.  If I paste in the format you suggest, I get a failure with the same message as above.

Reply:
Right, but before it is passed to `Specifiation.update_specification`, the [JSON object is loaded with `Parameters.read_json_revision`](https://github.com/PSLmodels/Cost-of-Capital-Calculator/blob/7422d73da462ffb6e47cc03d455f1d6b2b33d571/ccc/tests/test_parameters.py#L44). The object could alternatively get passed directly to `update_specification` because `parameters.ParamTools.adjust` accepts data as a Python `dict`, as a JSON string, or as a path to a JSON file.

Reply:
Also, I'm working on a better error message for these types of situations (https://github.com/PSLmodels/ParamTools/issues/61). That'll make JSON read errors clearer in the future.

Reply:
Ah - that's awesome that method handles a JSON string or dict!  Got it now.  Also, great work on clarifying the error messages.

Reply:
Thanks, our intern should have a PR up in the coming days.